### ISSUE TITLE
`catching polymorphic type` error with gcc 8.2.1

### ISSUE THREAD

**Issue by OTLabs** [openzim/zimwriterfs#88]
The Meson build system
Version: 0.48.2
Source dir: /home/titov/zimwriterfs-1.2
Build dir: /home/titov/zimwriterfs-1.2/build
Build type: native build
Project name: zimwriterfs
Project version: 1.2
Native C compiler: cc (gcc 8.2.1 "cc (GCC) 8.2.1 20180831")
Native C++ compiler: c++ (gcc 8.2.1 "c++ (GCC) 8.2.1 20180831")
Build machine cpu family: x86_64
Build machine cpu: x86_64
Dependency threads found: YES 
Found pkg-config: /usr/bin/pkg-config (1.5.4)
Dependency libzim found: YES 4.0.4
Dependency zlib found: YES 1.2.11
Dependency gumbo found: YES 0.10.1
Has header "magic.h": YES
Library magic found: YES
Configuring config.h using configuration
Build targets in project: 1
Option werror is: True [default: true]
Found ninja-1.8.2 at /usr/bin/ninja

ninja: Entering directory `build'
[3/6] Compiling C++ object 'src/src@@zimwriterfs@exe/tools.cpp.o'.
FAILED: src/src@@zimwriterfs@exe/tools.cpp.o 
c++ -Isrc/src@@zimwriterfs@exe -Isrc -I../src -I/usr/local/include -fdiagnostics-color=always -pipe -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -Wnon-virtual-dtor -Werror -std=c++11 -g -pthread  -MD -MQ 'src/src@@zimwriterfs@exe/tools.cpp.o' -MF 'src/src@@zimwriterfs@exe/tools.cpp.o.d' -o 'src/src@@zimwriterfs@exe/tools.cpp.o' -c ../src/tools.cpp
../src/tools.cpp: In function â€˜std::__cxx11::string getMimeTypeForFile(const string&)â€™:
../src/tools.cpp:493:19: error: catching polymorphic type â€˜class std::out_of_rangeâ€™ by value [-Werror=catch-value=]
     } catch (std::out_of_range) {}
                   ^~~~~~~~~~~~
../src/tools.cpp:499:17: error: catching polymorphic type â€˜class std::out_of_rangeâ€™ by value [-Werror=catch-value=]
   } catch (std::out_of_range) {}
                 ^~~~~~~~~~~~
cc1plus: all warnings being treated as errors
[5/6] Compiling C++ object 'src/src@@zimwriterfs@exe/mimetypecounter.cpp.o'.
ninja: build stopped: subcommand failed.

**Comment by OTLabs** [comment]
Got same error with gcc version 8.3.0.

**Comment by kelson42** [comment]
@OTLabs I can not reproduce the pb with gcc 8.2.0 on Ubuntu 19.04 with both git master `libzim` and `zimwriterfs`.

Compilation/Installation log of `libzim`:
```
kelson@camber:~/code/libzim$ rm -rf build && meson . build && ninja -C build && sudo ninja -C build install
The Meson build system
Version: 0.45.1
Source dir: /home/kelson/code/libzim
Build dir: /home/kelson/code/libzim/build
Build type: native build
WARNING: Newline character in a string detected, use ''' (three single quotes) for multiline strings instead.
This will become a hard error in a future Meson release.   description: 'Copy (or use mmap) header index buffers. (default:true) 10 15
Project name: libzim
Native C compiler: ccache cc (gcc 8.2.0 "cc (Ubuntu 8.2.0-1ubuntu2~18.04) 8.2.0")
Native C++ compiler: ccache c++ (gcc 8.2.0 "c++ (Ubuntu 8.2.0-1ubuntu2~18.04) 8.2.0")
Build machine cpu family: x86_64
Build machine cpu: x86_64
Checking for size of "off_t": 8
Found pkg-config: /usr/bin/pkg-config (0.29.1)
Native dependency zlib found: YES 1.2.11
Native dependency liblzma found: YES 5.2.2
Native dependency xapian-core found: YES 1.4.10
Dependency threads found: YES
Native dependency icu-i18n found: YES 60.2
Program libzim-compile-resources found: YES (/home/kelson/code/libzim/scripts/libzim-compile-resources)
Configuring config.h using configuration
Dependency GTest found: NO

Executing subproject gtest.

Project name: gtest
Native C++ compiler: ccache c++ (gcc 8.2.0 "c++ (Ubuntu 8.2.0-1ubuntu2~18.04) 8.2.0")
Dependency threads found: YES (cached)
Dependency threads found: YES (cached)
Dependency threads found: YES (cached)
Dependency threads found: YES (cached)
Build targets in project: 3

Subproject gtest finished.
Dependency gtest from subproject subprojects/gtest found: YES
Program cython3 found: NO
Native dependency python3 found: YES 3.6
Build targets in project: 8
Found ninja-1.8.2 at /home/kelson/bin/ninja
ninja: Entering directory `build'
[1/51] Generating resources with a custom command.
stopwords/en ['stopwords', 'en']
stopwords/he ['stopwords', 'he']
stopwords/fra ['stopwords', 'fra']
[51/51] Linking target test/cluster.
ninja: Entering directory `build'
[0/1] Installing files.
Installing src/libzim.so.4.0.5 to /usr/local/lib/x86_64-linux-gnu/libzim.so.4.0.5
Installing article.h to /usr/local/include/zim
Installing blob.h to /usr/local/include/zim
Installing error.h to /usr/local/include/zim
Installing file.h to /usr/local/include/zim
Installing fileheader.h to /usr/local/include/zim
Installing fileiterator.h to /usr/local/include/zim
Installing search.h to /usr/local/include/zim
Installing search_iterator.h to /usr/local/include/zim
Installing uuid.h to /usr/local/include/zim
Installing zim.h to /usr/local/include/zim
Installing article.h to /usr/local/include/zim/writer
Installing zimcreator.h to /usr/local/include/zim/writer
Installing /home/kelson/code/libzim/build/meson-private/libzim.pc to /usr/local/lib/x86_64-linux-gnu/pkgcon
```

Same for `zimwriterfs`
```
$ rm -rf build && meson . build && ninja -C build && sudo ninja -C build install
The Meson build system
Version: 0.45.1
Source dir: /home/kelson/code/zimwriterfs
Build dir: /home/kelson/code/zimwriterfs/build
Build type: native build
Project name: zimwriterfs
Native C compiler: ccache cc (gcc 8.2.0 "cc (Ubuntu 8.2.0-1ubuntu2~18.04) 8.2.0")
Native C++ compiler: ccache c++ (gcc 8.2.0 "c++ (Ubuntu 8.2.0-1ubuntu2~18.04) 8.2.0")
Build machine cpu family: x86_64
Build machine cpu: x86_64
Dependency threads found: YES
Found pkg-config: /usr/bin/pkg-config (0.29.1)
Native dependency libzim found: YES 4.0.5
Native dependency zlib found: YES 1.2.11
Native dependency gumbo found: YES 0.10.1
Has header "magic.h": YES
Library magic found: YES
Configuring config.h using configuration
Build targets in project: 1
Found ninja-1.8.2 at /home/kelson/bin/ninja
ninja: Entering directory `build'
[6/6] Linking target src/zimwriterfs.
ninja: Entering directory `build'
[0/1] Installing files.
Installing src/zimwriterfs to /usr/local/bin/zimwriterfs
```

**Comment by OTLabs** [comment]
@kelson42 Thank you for taken a look at an issue I experience. I use Alpine Linux (Edge version) to build packages for libzim and zimwriterfs. I have successfully built libzim v4.0.5 package, but zimwriterfs v1.2 package produced the last error. I assume that there is no much difference between v1.2 release and current master branch, am I right? Thank you for provided logs, hope I will be able to nail down the origin of the problem I experience.

**Comment by OTLabs** [comment]
Looks like this issue was fixed in this commit a9de187522613c5dca306dcef527bc913ccd0ee8 as a fix for #64.

Would it be possible to create new release for zimwriterfs so that commit would be included?

Thank you in advance!

**Comment by kelson42** [comment]
@OTLabs Thx for reporting and checking that if works fine now. A new release of zimwriterfs should endeed be done soon. Probably next week.
